crates/core/src/operations/unset_tbl_properties.rs

```diff
@@ -0,0 +1,169 @@
+        Self {
+            properties: None,
+            raise_if_not_exists: true,
+            snapshot,
+            log_store,
+            app_metadata: None,
+        }
+    }
+
+    /// Specify the constraint to be removed
+    pub fn with_property(mut self, name: String) -> Self {
+        match self.properties {
+            Some(mut vec_name) => {
+                vec_name.push(name);
+                self.properties = Some(vec_name);
+            }
+            None => self.properties = Some(vec![name]),
+        }
+        self
+    }
+
+    /// Specify the constraint to be removed
+    pub fn with_properties(mut self, names: Vec<String>) -> Self {
+        match self.properties {
+            Some(mut vec_name) => {
+                vec_name.extend(names);
+                self.properties = Some(vec_name);
+            }
+            None => self.properties = Some(names),
+        }
+        self
+    }
+
+    /// Specify if you want to raise if the constraint does not exist
+    pub fn with_raise_if_not_exists(mut self, raise: bool) -> Self {
+        self.raise_if_not_exists = raise;
+        self
+    }
+
+    /// Additional metadata to be added to commit info
+    pub fn with_metadata(
+        mut self,
+        metadata: impl IntoIterator<Item = (String, serde_json::Value)>,
+    ) -> Self {
+        self.app_metadata = Some(HashMap::from_iter(metadata));
+        self
+    }
+}
+
+impl std::future::IntoFuture for UnsetTablePropertiesBuilder {
+    type Output = DeltaResult<DeltaTable>;
+
+    type IntoFuture = BoxFuture<'static, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        let mut this = self;
+
+        Box::pin(async move {
+            let properties = match this.properties {
+                Some(v) => v,
+                None => {
+                    return Err(DeltaTableError::Generic(
+                        "No properties provided".to_string(),
+                    ))
+                }
+            };
+
+            let mut metadata = this.snapshot.metadata().clone();
+
+            // Check if names are valid config keys
+            for key in &properties {
+                DeltaConfigKey::from_str(key)?;
+            }
+
+            let mut incorrect_config_names: Vec<&str> = Vec::new();
+            for key in &properties {
+                if !metadata.configuration.contains_key(key) {
+                    incorrect_config_names.push(key)
+                }
+            }
+
+            if this.raise_if_not_exists && incorrect_config_names.len() > 0 {
+                return Err(DeltaTableError::Generic(format!(
+                    "table properties with names: {:?} don't exists",
+                    incorrect_config_names
+                )));
+            } else if !this.raise_if_not_exists && incorrect_config_names.len() == properties.len()
+            {
+                return Ok(DeltaTable::new_with_state(this.log_store, this.snapshot));
+            }
```

Comment:
could we check (or did we already) how spark behaves here? while this is a noop, it may be that we still want to create a commit to recognise that the operation was executed, but not sure.

Reply:
Sure, I'll check this one over the weekend!

Reply:
@roeap sorry for the very late response :) But Spark-delta still does the commit, so I will update the logic to do that as well